pkg/cloudprovider/openstack.go

```diff
@@ -487,7 +488,7 @@ func (o *OpenStack) getNeutronPortNodeEgressIPConfiguration(p neutronports.Port)
 
 	}
 
```

Comment:
It's a bit odd in OSP, but I decided to report the capacity per OSP interface.  So, pass in cloudPrivateIPConfigs, and then in line 490 make sure that the cloudPrivateIP is acutally part of the ipnet:
~~~
        var cloudPrivateIPsCount int

	// Loop over all subnets. OpenStack potentially has several IPv4 or IPv6 subnets per port, but the
	// CloudPrivateIPConfig expects only a single subnet of each address family per port. Throw an error
	// in such a case.
	for _, s := range subnets {
		// Parse CIDR information into ip and ipnet.
		ip, ipnet, err = net.ParseCIDR(s.CIDR)
		if err != nil {
			return nil, fmt.Errorf("could not parse subnet information %s for network %s, err: %q",
				s.CIDR, p.NetworkID, err)
		}
		// For IPv4 and IPv6, calculate the capacity.
		if utilnet.IsIPv4(ip) {
			if ipv4 != "" {
				return nil, fmt.Errorf("found multiple IPv4 subnets attached to port %s, this is not supported", p.ID)
			}
			ipv4 = ipnet.String()
		} else {
			if ipv6 != "" {
				return nil, fmt.Errorf("found multiple IPv6 subnets attached to port %s, this is not supported", p.ID)
			}
			ipv6 = ipnet.String()
		}
                // Loop over all cloudPrivateIPConfigs and check if they are part of this ipnet.
                // If the IP is contained in the ipnet, increase the counter.
                for _, cpic := range cloudPrivateIPConfigs {
                  cip, _ := cloudprivateipconfig.CloudPrivateIPConfigNameToIP(cpic.name)
                  if ipnet.Contains(cip) {
                    cloudPrivateIPsCount++
                 }
                }
	}

        c := openstackMaxCapacity + cloudPrivateIPsCount - len(p.AllowedAddressPairs)
~~~
something like that

Reply:
sure, done.